graph.cpp

```diff
@@ -27,13 +27,29 @@ Graph::Graph(uint64_t num_nodes): num_nodes(num_nodes) {
   GraphWorker::startWorkers(this, bf);
 }
 
+Graph::Graph(std::ifstream& file) {
+  file >> num_nodes;
+#ifdef VERIFY_SAMPLES_F
+  cout << "Verifying samples..." << endl;
+#endif
+  representatives = new set<Node>();
+  supernodes = new Supernode*[num_nodes];
+  parent = new Node[num_nodes];
+  for (Node i=0;i<num_nodes;++i) {
+    representatives->insert(i);
+    supernodes[i] = new Supernode(num_nodes, file);
+    parent[i] = i;
+  }
+  is_copy = true;
```

Comment:
Same with this variable. There are a number of places where we do different things for copies. How would that fit into what we want to do with this.